metrics/helpers.go

```diff
@@ -0,0 +1,64 @@
+package metrics
+
+import (
+	"errors"
+	"fmt"
+
+	dto "github.com/prometheus/client_model/go"
+)
+
+// MatchesLabels returns true if m has all the labels in l.
+func MatchesLabels(m *dto.Metric, labelNamesAndValues ...string) bool {
+	for _, l := range makeLabels(labelNamesAndValues...) {
```

Comment:
nit: do we need to create `[]*dto.LabelPair` first? If not, we can save some work and allocations here.

Reply:
We don't have to, but it makes the code below clearer as we can refer to things as `Name` and `Value` rather than indexing into an array. Given this method is only intended to be used in tests, and with a small number of labels, I don't think the allocations are an issue, but happy to revisit this if you disagree.

(Given your approval, I'm going to merge the PR, but happy to address this in a follow-up PR.)

Reply:
> Given this method is only intended to be used in tests, and with a small number of labels, I don't think the allocations are an issue, but happy to revisit this if you disagree.

I see. We can improve it once it becomes part of critical path :)